ql/src/java/org/apache/hadoop/hive/ql/txn/compactor/Cleaner.java

```diff
@@ -97,23 +46,28 @@ public class Cleaner extends MetaStoreCompactorThread {
   static final private Logger LOG = LoggerFactory.getLogger(CLASS_NAME);
   private boolean metricsEnabled = false;
 
-  private ReplChangeManager replChangeManager;
   private ExecutorService cleanerExecutor;
+  private List<RequestHandler> requestHandlers;
 
   @Override
   public void init(AtomicBoolean stop) throws Exception {
     super.init(stop);
-    replChangeManager = ReplChangeManager.getInstance(conf);
     checkInterval = conf.getTimeVar(
             HiveConf.ConfVars.HIVE_COMPACTOR_CLEANER_RUN_INTERVAL, TimeUnit.MILLISECONDS);
-    cleanerExecutor = CompactorUtil.createExecutorWithThreadFactory(
-            conf.getIntVar(HiveConf.ConfVars.HIVE_COMPACTOR_CLEANER_THREADS_NUM),
-            COMPACTOR_CLEANER_THREAD_NAME_FORMAT);
     metricsEnabled = MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.METRICS_ENABLED) &&
         MetastoreConf.getBoolVar(conf, MetastoreConf.ConfVars.METASTORE_ACIDMETRICS_EXT_ON);
-    boolean tableCacheOn = MetastoreConf.getBoolVar(conf,
-            MetastoreConf.ConfVars.COMPACTOR_CLEANER_TABLECACHE_ON);
-    initializeCache(tableCacheOn);
+    if (CollectionUtils.isEmpty(requestHandlers)) {
```

Comment:
Why do we need this check? it's a bit confusing initialization. I would assume this is required for tests?

Reply:
Yes this is required for tests